keras_cv/layers/hierarchical_transformer_encoder.py

```diff
@@ -0,0 +1,67 @@
+import math
+
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.layers.efficient_multihead_attention import (
+    EfficientMultiheadAttention,
+)
+from keras_cv.layers.regularization.drop_path import DropPath
+
+
+@keras.saving.register_keras_serializable(package="keras_cv")
+class HierarchicalTransformerEncoder(keras.layers.Layer):
+    def __init__(
+        self,
+        project_dim,
+        num_heads,
+        sr_ratio=1,
+        drop_prob=0.0,
+        layer_norm_epsilon=1e-6,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.norm1 = keras.layers.LayerNormalization(epsilon=layer_norm_epsilon)
+        self.attn = EfficientMultiheadAttention(
+            project_dim, num_heads, sr_ratio
+        )
+        self.drop_path = DropPath(drop_prob)
+        self.norm2 = keras.layers.LayerNormalization(epsilon=layer_norm_epsilon)
+        self.mlp = self.MixFFN(
+            channels=project_dim,
+            mid_channels=int(project_dim * 4),
+        )
+
+    def build(self, input_shape):
```

Comment:
When overriding build we should also either call `super().build` or just set `self.built = True`

Reply:
Nice catch!